### PR TITLE
model: add ChangeStateInfo to ColumnInfo

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve? 
This PR is used to support column type changing.

### What is changed and how it works?
Add ChangeStateInfo to ColumnInfo.

### CODE REVIEW

File: model/model.go [hunk 1]
```diff
@@ -96,7 +102,8 @@ type ColumnInfo struct {
 	State               SchemaState `json:"state"`
 	Comment             string      `json:"comment"`
 	// A hidden column is used internally(expression index) and are not accessible by users.
-	Hidden bool `json:"hidden"`
+	Hidden           bool `json:"hidden"`
+	*ChangeStateInfo `json: "change_state_info"`
```
Comment (kennytm) [inline model/model.go:106]:
```suggestion
	*ChangeStateInfo `json:"change_state_info"`
```

Comment (zimulala) [reply model/model.go]: Done.

**Comment by zimulala** [comment]
PTAL @kennytm @tangenta

**Comment by zimulala** [comment]
PTAL @AilinKid  @tangenta

**Pull request merged**